lib/minitag/minitest_tag.rb

```diff
@@ -36,6 +36,21 @@ def tag(*tags)
       Minitag.register_for_extension(self)
     end
 
+    def run_one_method(klass, method_name, reporter) # rubocop:disable Metrics/AbcSize
```

Comment:
I'm going over the source code and this seems to achieve what we need here, but it relies a lot on how the internals are laid out. I'm wondering if instead of forcing a skip we could create a new reporter that would report just the skipped tests.

Don't take this as an opinion yet, I'm in the brainstorming phase. :)

Reply:
That's a valid critique: building that `Skip` object is not the most beautiful code to behold :stuck_out_tongue: . I'm not sure that a custom reporter will do what we want, though. The whole point of this is to show the user that these tests were skipped, using the reporters they configured (I personally use some from minitest-reporters). Adding our own reporter into the mix doesn't really do that... does it? It seems like we need to hand minitest a result such that existing reporters can interpret it. There might be another way to do this though: what if we redefine the test methods on the fly to invoke the `skip` function? I'll investigate that a litle.

Reply:
Yeah that actually seems to work pretty well. I pushed an update, see what you think. The `run_one_method` has entirely disappeared, and the meat of the logic has moved to [here](https://github.com/bernardoamc/minitag/pull/19/files#diff-750b4db13894053d0a66158e0ea2b9e1014cf54a3e01571db40f8e007d585363R65-R70).

Reply:
Another share of ugly with this approach is performance. It's actually _significantly_ slower, expect in part because of the patching, and in part because patching in a new method re-run's minitag's hook code. Here is a run through my codebase with 816 tests with the initial version of this PR:

```
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Finished in 0.090732s, 8993.5422 runs/s, 0.0000 assertions/s.
```

Here is a run with the current version:

```
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Finished in 10.412197s, 78.3696 runs/s, 10.0843 assertions/s.

816 runs, 105 assertions, 0 failures, 0 errors, 816 skips
```

Nearly 10x slower. I mean, it's still only ten seconds, I can live with that, but some codebases will be even larger. Something to consider.

Reply:
I agree with you, 10x performance hit seems to not be worth it for what is happening. 😄 I can't investigate it now, but I have a hunch that we can can redefine the `run` method of `klass.new(method_name)` to return a Result instead of overriding the entire `run_one_method`.

Reply:
Yeah we can, but as far as I can tell that would really only save us from actually recording the result in the reporter. We would still need to create the `Skip` object, because that's what is used to determine [if the test was actually skipped](https://github.com/minitest/minitest/blob/a9fa045044b4210cfd21a512b06d1a4527d709ba/lib/minitest.rb#L517-L519). Would you like me to revert the latest commit and move the `Skip` creation code into there?

Reply:
Let's go with your original approach since we know it works and keep an eye open for better alternatives. No reason to block this feature since it will help you. :)

Reply:
Alright, I ripped that last commit off: it's back to the original approach.